postgrid/__init__.py

```diff
@@ -495,10 +624,35 @@ def _pm_convert_json_value(value):
             new_value[_camel_to_snake(key)] = inner_value
             continue
 
-        if key != 'metadata' and isinstance(inner_value, dict) and 'object' in inner_value:
-            new_value[_camel_to_snake(key)] = _pm_convert_json_value(inner_value)
+        if (
+            key != 'metadata'
+            and isinstance(inner_value, dict)
+            and 'object' in inner_value
+        ):
+            new_value[_camel_to_snake(
+                key)] = _pm_convert_json_value(inner_value)
             continue
 
         new_value[key] = inner_value
 
-    return PM_OBJECT_TO_CLASS[new_value['object']](**_map_keys_recursive(new_value, _camel_to_snake))
+    return PM_OBJECT_TO_CLASS[new_value['object']](
+        **_map_keys_recursive(new_value, _camel_to_snake)
+    )
+
+
+def _av_convert_json_value(value):
+    new_value = {}
+    for key, inner_value in value.items():
+        if isinstance(inner_value, dict):
+            new_value[_camel_to_snake(
+                key)] = _av_convert_json_value(inner_value)
+            continue
+        elif isinstance(inner_value, list):
+            for i in range(len(inner_value)):
+                if isinstance(inner_value[i], dict):
+                    inner_value[i] = _av_convert_json_value(inner_value[i])
+            new_value[_camel_to_snake(key)] = inner_value
+            continue
+        new_value[key] = inner_value
+
+    return Address(**_map_keys_recursive(new_value, _camel_to_snake))
```

Comment:
Create different classes for different endpoints if possible.

Reply:
Might be a bit of a difficult task with the current AV api since PM returns the object type. Not sure if we want to add something similar to AV?

Reply:
Ah misread the previous message, added separate classes for the endpoints